subsys/bluetooth/host/scan.c

```diff
@@ -516,6 +528,8 @@ void bt_hci_le_adv_ext_report(struct net_buf *buf)
 	while (num_reports--) {
 		struct bt_hci_evt_le_ext_advertising_info *evt;
 		struct bt_le_scan_recv_info adv_info;
+		uint8_t size_to_copy;
+		bool truncate;
```

Comment:
As mentioned earlier, I think it's better to discard incomplete data than sending it to the upper layers. 

I'll not review the code handling truncation, unless you strongly disagree and want to keep this part.

Reply:
We still have to keep track of truncation in the host as we may need to discard the remaining parts of a report coming from the controller.

Reply:
We do indeed need to keep track of when/if the data is larger than our buffer. I had briefly forgotten that, unlike ISO, adv data does not have a `start` data status, so we do check if it too large (not sure I would call that "truncate" though), as if it is too large, we just need to discard any incoming until it is complete.

Reply:
Yes. The fact that the spec does not indicate `start` makes the implementation slightly more complex.